packages/mui-utils/src/getElementRef/getElementRef.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+/**
+ * Returns the ref of an element handling differences between React 19 and older versions
+ *
+ * @param element React.ReactElement<any>
+ * @returns React.Ref<any> | null
+ */
+export default function getElementRef(element: React.ReactElement<any>): React.Ref<any> | null {
+  if (!element || !React.isValidElement(element)) {
+    return null;
+  }
+
+  // 'ref' is passed as prop in React 19, whereas 'ref' is directly attached to children in older versions
+  return (element.props as any).propertyIsEnumerable('ref')
+    ? (element.props as any).ref
+    : // @ts-expect-error element.ref is not included in the ReactElement type
+      // We cannot check for it, but isValidElement is true at this point
```

Comment:
`element.ref` does not exist in ReactElement. We've silenced this error in our codebase since: https://github.com/mui/material-ui/pull/24565/files#diff-dcdcbff6b975b552227833f729bf59d31e97e4876a9d6ce982a07c9e1436d9bfR90

Here are other occurrences, probably copy-pasted: https://github.com/search?q=repo%3Amui%2Fmaterial-ui%20%22TODO%20upstream%20fix%22&type=code

I don't think they'll fix it now that they moved the ref to the props.

Reply:
I would just recommend if you find an open issue about this, to link in the comment above so we know once it's merged that we can remove the `@ts-ignore`.

Reply:
I couldn't find any issue in [the original PR](https://github.com/mui/material-ui/pull/24565), [DefinitelyTyped issues](https://github.com/DefinitelyTyped/DefinitelyTyped/issues), or [DefinitelyTyped discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions?discussions_q=reactelement&page=1). I don't know if they have discussed it previously. I don't think it will get much traction as it's probably been like this forever and it's now removed in React 19.

I created a [discussion asking for this](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/70189) and linked it.